### ISSUE TITLE
Context switch trace on 32-bit ARM Linux doesn't capture PIDs correctly

### ISSUE THREAD

**Issue by zyp** [wolfpld/tracy#328]
When doing a context switch trace on 32-bit ARM Linux, I'm getting bogus PIDs in the capture. After digging into the issue, I realized that the values I'm getting are not the PIDs from `next_pid`, but priority values, from the neighboring `next_prio` field.

Excerpt from `TracySysTrace.cpp`:

```cpp
                            // Data (not ABI stable, but has not changed since it was added, in 2009):
                            //   u8  hdr[8]
                            //   u8  prev_comm[16]
                            //   u32 prev_pid
                            //   u32 prev_prio
                            //   u64 prev_state
                            //   u8  next_comm[16]
                            //   u32 next_pid
                            //   u32 next_prio
```

I suspect that the size of the `prev_state` field is platform dependent, and the following workaround does appear to solve the problem for me:

```diff
diff --git a/client/TracySysTrace.cpp b/client/TracySysTrace.cpp
index 40f947ca..ec2209c7 100644
--- a/client/TracySysTrace.cpp
+++ b/client/TracySysTrace.cpp
@@ -1322,8 +1322,8 @@ void SysTraceWorker( void* ptr )
 
                             ring.Read( &prev_pid, offset, sizeof( uint32_t ) );
                             offset += sizeof( uint32_t ) + sizeof( uint32_t );
-                            ring.Read( &prev_state, offset, sizeof( uint64_t ) );
-                            offset += sizeof( uint64_t ) + 16;
+                            ring.Read( &prev_state, offset, sizeof( uint32_t ) );
+                            offset += sizeof( uint32_t ) + 16;
                             ring.Read( &next_pid, offset, sizeof( uint32_t ) );
 
                             uint8_t reason = 100;
```

**Comment by wolfpld** [comment]
`prev_state` is indeed `long`, which is 32-bit on LP32/ILP32, but 64 bit on LP64.

**Comment by wolfpld** [comment]
Fixed in 46d29776.
